jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/NoOpSession.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.apphosting.runtime.jetty9;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpSession;
+import javax.servlet.http.HttpSessionContext;
+
+/**
+ * A minimal implementation of HttpSession, just sufficient to use attributes within a
+ * single request.
+ */
+public class NoOpSession implements HttpSession {
+
+  protected ServletContext context;
+  protected String clusterId;
+  protected String nodeId;
+  protected boolean valid = true;
+  private final Map<String, Object> attributes = new HashMap<>();
+
+  /**
+   * @param id the identity of the session
+   * @param nodeId the id of the session qualified with the node workername
+   * @param context the ServletContext to which the session belongs
+   */
+  protected NoOpSession(String id, String nodeId, ServletContext context) {
+    this.clusterId = id;
+    this.nodeId = nodeId;
+    this.context = context;
+  }
+
+  /**
+   * Finish a request
+   */
+  protected void complete() {
+    attributes.clear();
+  }
+
+  @Override
+  public long getCreationTime() {
+    checkValid();
+    return 0;
+  }
+
+  @Override
+  public String getId() {
+    return clusterId;
```

Comment:
Looks like I missed checkValid() on set/removeAttribute(), so I'll add those in. Otherwise I've doublechecked with the servlet spec api javadoc and verified that we are doing checkValid() on the correct methods.